Automattic/vip-design-system
# TS migration: Moves spinner component to TS

## Description

This Pull request is part of hackathon from Connoisseurs Rome Meetup 2023 and it moves the component Spinner to TS.

## Checklist


## Steps to Test

1. pull down the PR
2. run `npm run start`
3. Check storybook on localhost:6006 to validate that the spinner is same as the storybook on production/trunk
